ca/certificates.go

```diff
@@ -410,8 +423,22 @@ func NewRootCA(certBytes, keyBytes []byte, certExpiry time.Duration) (RootCA, er
 
 func ensureCertKeyMatch(cert *x509.Certificate, key crypto.PublicKey) error {
 	switch certPub := cert.PublicKey.(type) {
-	// TODO: Handle RSA keys.
+	case *rsa.PublicKey:
+		if certPub.N.BitLen() < 2048 {
+			return errors.New("RSA keys must be >= 2048 bits")
+		}
+		rsaKey, ok := key.(*rsa.PublicKey)
+		if ok && certPub.E == rsaKey.E && certPub.N.Cmp(rsaKey.N) == 0 {
```

Comment:
Since we're insisting on a certain key size, should we also do additional checks on the RSA key like validating that `E` has a reasonable value, or is this just silly?